File: cpp_src/MultiplyStrings.cpp
class Solution {
public:
    string multiply(string num1, string num2) {
        int size1 = num1.size();
        int size2 = num2.size();
        int size = size1 + size2;
        if(size == 0) return "0";
        vector<int> vect1(size1,0);
        vector<int> vect2(size2,0);
        for(int i = 0; i < size1; i++){
            vect1[size1 - 1 -i] = num1[i] - '0';
        }
        for(int i = 0; i < size2; i++){
            vect2[size2 - 1 - i] = num2[i] - '0';
        }
        vector<int> nums(size,0);
        for(int i = 0; i < size1; i++){
            for(int j = 0; j < size2; j++){
                int value = vect1[i] * vect2[j];
                nums[i + j] += value % 10;
                nums[i + j + 1] += value / 10;
            }
        }
        for(int i = 0; i < size - 1; i++){
            if(nums[i] >= 10){
                nums[i + 1] += nums[i] / 10;
                nums[i] %= 10;
            }
        }
        string result;
        int index = size - 1;
        while(nums[index] == 0 && index > 0) index--;
        for(int i = index; i >= 0; i--){
            result.append(1,nums[i] + '0');
        }
        return result;
    }
};

File: python_src/MultiplyStrings.py
class Solution(object):
    def multiply(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        size1 = len(num1)
        size2 = len(num2)
        
        num1 = [int(x) for x in num1]
        num2 = [int(x) for x in num2]
        num1.reverse()
        num2.reverse()
        result = [0] * (size1 + size2)
        for i in range(size1):
            for j in range(size2):
                temp = num1[i] * num2[j]
                result[i + j + 1] += temp / 10
                result[i + j] += temp % 10
        for i, value in enumerate(result):
            try:
                result[i + 1] += value / 10
                result[i] = value % 10
            except:
                pass
        while len(result) > 1 and result[-1] == 0: del result[-1]
        result.reverse()
        return "".join([str(x) for x in result])
        
        